docs/DeveloperGuide.md

```diff
@@ -233,6 +233,40 @@ From the notes in the sequence diagram above, for every attribute in the instrum
 prompt to get user to provide information for that attribute. This is done through a series of methods in
 the `TextUi` class.
 
+### Edit instrument feature
+The edit instrument functionality mainly involves the `console`, `commands` and `model` components. Within the `console`
+component, the `InputParser` class implements the method `InputParser#getEditInstrumentCommand()`. This method calls
+`InputParser#getParametersToEdit` which will prompt the users to input which parameters of the instrument to edit
+and check if the parameters entered are valid. Invalid inputs will not be processed.
+
+The process of writing the new values of the parameters to be edited is handled by the `EditInstrumentParser` class.
+The method `EditInstrumentParser#createEditCommand()` calls `EditInstrumentParser#getEditedParameters()` which 
+calls multiple individual methods that check if its parameters is being edited and to enter a new value for the 
+parameters.
+
+The figure below represents the sequence diagram when the user wants to edit the name a stock:
+
+<img src="images/EditInstrumentSequenceDiagram.png" width="600"/>
+
+More details about the reference frame for checking if a parameters is entered and to add to HashMap if entered is given below:
+
+<img src="images/EditRefrence.png" width="600"/>
+
+More details about the execution of setting the stock with the new values is shown below: (in this case is setting the name parameter to new name)
+
+<img src="images/EditExecuteSequenceDiagram.png" width="600"/>
```

Comment:
Make it bigger, make sure the text in diagram closely matches the normal text in dg, same for all the diagrams